### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by jsoref**
While working on https://github.com/jenkinsci/jenkins/pull/4407#discussion_r369667658, I made a mistake in not recognizing that the default `withMode` is [different](https://github.com/jenkinsci/remoting/compare/master...jsoref:cleanup?expand=1#diff-8a0e2be489e004b87cc21e1fc8d17d3dR407) between `Channel(...)` and `ChannelBuilder(...)`.

I'm not absolutely certain about my comments, but if they're correct, it would help the next person like me perform a similar migration.

### CODE REVIEW

File: src/main/java/hudson/remoting/Channel.java [hunk 1]
```diff
@@ -1865,7 +1916,7 @@ public static void dumpDiagnosticsForAll(@Nonnull PrintWriter w) {
      * Notification that {@link Command#readFrom} has succeeded.
      * @param cmd the resulting command
      * @param blockSize the serialized size of the command
-     * @see CommandListener
```
Comment (jsoref) [inline src/main/java/hudson/remoting/Channel.java]: IntelliJ and everything else, myself included, were absolutely confused about this reference.

File: src/main/java/hudson/remoting/SynchronousCommandTransport.java [hunk 2]
```diff
@@ -57,14 +57,14 @@ public ReaderThread(CommandReceiver receiver) {
         public void run() {
             final String name =channel.getName();
             try {
-                while(!channel.isInClosed()) {
+                while (!channel.isInClosed()) {
```
Comment (jsoref) [inline src/main/java/hudson/remoting/SynchronousCommandTransport.java]: These whitespace changes were enforced by IntelliJ when I made an unrelated debugging change (which is not included in this PR).

File: src/main/java/hudson/remoting/Channel.java [hunk 3]
```diff
@@ -1618,7 +1665,9 @@ public OutputStream getUnderlyingOutput() {
      *      The remote port that the connection will be forwarded to.
      * @return
      *      Created {@link PortForwarder}
+     * @deprecated as of 3.39
```
Comment (jsoref) [inline src/main/java/hudson/remoting/Channel.java]: The `PortForwarder` class was Deprecated in #361, but no one bubbled up the deprecation within this module.
@jvz @jeffret-b

File: src/main/java/hudson/remoting/Channel.java [hunk 4]
```diff
@@ -464,19 +494,28 @@ public Channel(String name, ExecutorService exec, Mode mode, InputStream is, Out
...
      * @since 2.13
      */
     @Deprecated
     public Channel(String name, ExecutorService exec, CommandTransport transport, boolean restricted, ClassLoader base) throws IOException {
         this(new ChannelBuilder(name,exec)
                 .withBaseLoader(base)
-                .withRestricted(restricted), transport);
+                .withArbitraryCallableAllowed(!restricted)
+                .withRemoteClassLoadingAllowed(!restricted)
+                , transport);
```
Comment (jsoref) [inline src/main/java/hudson/remoting/Channel.java]: This style is used elsewhere within this file.

File: src/main/java/hudson/remoting/Channel.java [hunk 5]
```diff
@@ -464,19 +494,28 @@ public Channel(String name, ExecutorService exec, Mode mode, InputStream is, Out
...
+     *                  .withRemoteClassLoadingAllowed(!restricted)
+     *                  .withBaseLoader(base)
+     *                  .build(transport)
      * @since 2.13
      */
     @Deprecated
     public Channel(String name, ExecutorService exec, CommandTransport transport, boolean restricted, ClassLoader base) throws IOException {
         this(new ChannelBuilder(name,exec)
                 .withBaseLoader(base)
-                .withRestricted(restricted), transport);
```
Comment (jsoref) [inline src/main/java/hudson/remoting/Channel.java]: `withRestricted` is deprecated.

**Comment by jsoref** [comment]
I'm not sure why Jenkins doesn't like my PRs of late. The changes here are really trivial. They should pass tests. I'm going to force push a couple of times until I have a sense of what's going on. Please don't review until I've figured out what's going on.

**Comment by jsoref** [comment]
I'm done pushing to this PR. There was nothing wrong w/ any of my commits, the CI was flaky.

**Comment by jeffret-b** [comment]
Thanks for the PRs. It will probably be a few days before I can get to them. Anyone else is also welcome to review.
